src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/AdditionalFilesItemHandler.cs

```diff
@@ -12,22 +12,30 @@ namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices.Handlers
     /// <summary>
     ///     Handles changes to the  &lt;AdditionalFiles/&gt; item during design-time builds.
     /// </summary>
+    [Export(typeof(IWorkspaceContextHandler))]
     internal class AdditionalFilesItemHandler : ICommandLineHandler
     {
         // WORKAROUND: To avoid Roslyn throwing when we add duplicate additonal files, we remember what 
         // sent to them and avoid sending on duplicates.
         // See: https://github.com/dotnet/project-system/issues/2230
 
         private readonly UnconfiguredProject _project;
-        private readonly IWorkspaceProjectContext _context;
```

Comment:
seems like a lot of logic could be shared inside an abstract class regarding `_context`.  At lease I'd like to change `_context` to be a property so we throw a descriptive exception when someone accesses it without calling Initialize.  Needing to remember to check for null on each access and throw seems like a lot of boilerplate